build.rs

```diff
@@ -39,11 +39,13 @@ fn channel() -> String {
 }
 
 fn commit_hash() -> Option<String> {
-    Command::new("git")
-        .args(["rev-parse", "--short", "HEAD"])
+    let output = Command::new("git")
+        .args(["rev-parse", "HEAD"])
         .output()
-        .ok()
-        .and_then(|r| String::from_utf8(r.stdout).ok())
+        .ok()?;
+    let mut stdout = String::from_utf8(output.stdout).ok()?;
+    stdout.truncate(10);
```

Comment:
Is there a reason we're truncating to 10? 

I checked the hash length for some other tools and they're either 8-9 characters long:

```
rustc 1.81.0-nightly (6b0f4b5ec 2024-06-24)
cargo 1.81.0-nightly (bc89bffa5 2024-06-22)
clippy 0.1.81 (6b0f4b5e 2024-06-24)
rustup 1.27.1 (54dd3d00f 2024-04-24)
rust-analyzer 1.81.0-nightly (6b0f4b5e 2024-06-24)
```

Also, I think some of the non-determinism came from using the `--short` option with git's `rev-parse` subcommand. Here are the docs:

```
       --short[=length]
           Same as --verify but shortens the object name to a unique prefix with at least length
           characters. The minimum length is 4, the default is the effective value of the
           core.abbrev configuration variable (see git-config(1))
```

Reply:
The rust-lang/rust repo (which rustfmt releases are published from, and therefore the one whose commit hashes matter) currently has about 262k commits on master and 398k commits reachable from PR branches. Using 800k in the estimations to account for some growth, the formula for probability of any colliding _H_-digit hash over _N_ years is:

$$ 1 - \left( 1 - \frac{800{\small,}000}{16^{H}} \right)^{\Huge \raise -1ex {429 N}} $$

where 429 is the estimated number of releases per year, including 365 nightlies, 8 stables, and 7 betas per stable release. For 8-digit hash, there is 33% probability of a collision in 5 years. For 9-digit, 2.5%, and 10-digit, 0.16%.

Reply:
Thank you for the detailed and thoughtful explanation!